server/src/main/java/io/crate/expression/predicate/NotPredicate.java

```diff
@@ -226,6 +226,12 @@ public Query toQuery(Function input, LuceneQueryBuilder.Context context) {
                 var refExistsQuery = IsNullPredicate.refExistsQuery(nullableRef, context, false);
                 if (refExistsQuery != null) {
                     builder.add(refExistsQuery, BooleanClause.Occur.MUST);
+                } else {
+                    // fall back to less efficient query
+                    return new BooleanQuery.Builder()
+                        .add(notX, Occur.MUST)
+                        .add(LuceneQueryBuilder.genericFunctionFilter(input, context), Occur.FILTER)
+                        .build();
```

Comment:
The tests added do not touch this path but added for completeness, if `refExistsQuery` is null, we should fall back.